src/ethereum/mod.rs

```diff
@@ -218,6 +272,30 @@ impl Ethereum {
                 )
             })
     }
+
+    /// Register a block observer.
+    ///
+    /// # Arguments
+    ///
+    /// * `observer` - Any object which implements observer traits
+    ///
+    pub fn register_observer(&mut self, observer: Reactor) {
+        self.observers.push(observer);
+    }
+
+    /// Notify all the block observers
+    ///
+    /// # Arguments
+    ///
+    /// * `block` - block to notify
+    ///
+    pub fn notify_all_observers(&mut self, block: &Block) {
```

Comment:
Is it correct that this function is public? Where is it used outside this module?

Reply:
It's called from 'observer' module